### PR TITLE
Read npmrc in constructor

### LINKED ISSUE #58: Read npmrc in constructor
If no other auth is available.

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes #58

### CODE REVIEW

File: npm.js [hunk 1]
```diff
@@ -167,35 +163,25 @@ function configureCredentials(registry, _auth, ui) {
...
-    if (e.code == 'ENOENT')
-      return;
-    throw e;
-  })
-  .then(function(npmrc) {
-    if (npmrc)
+  return new Promise(nprc.exists)
+  .then(function(npmrcExists) {
+    if (npmrcExists)
       return ui.confirm('npmrc found, would you like to use these settings?', true)
```
Comment (guybedford) [inline npm.js]: I don't think we need this at all any more?

**Comment by guybedford** [comment]
Looks good.

File: npm.js (hunk 1 above)
Comment (crisptrutski) [inline npm.js]: Yeah, we'll replace this with #68

File: lib/npmrc.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+var asp = require('rsvp').denodeify;
+var fs = require('graceful-fs');
+var path = require('path');
+var utils = require('./utils');
+
+function getFilepath() {
+  var home = process.env.HOME || process.env.USERPROFILE || process.env.HOMEPATH;
```
Comment (guybedford) [inline lib/npmrc.js]: Are you sure this is exactly the same resolution npm uses?

File: npm.js [hunk 3]
```diff
@@ -63,23 +65,13 @@ function clone(a) {
...
-}
-
 var NPMLocation = function(options, ui) {
+  var npmrc = new Npmrc();
+
   this.ui = ui;
   this.name = options.name;
   // default needed during upgrade time period
-  this.registryURL = options.registry || defaultRegistry;
+  this.registryURL = options.registry || npmrc.getRegistry() || defaultRegistry;
```
Comment (guybedford) [inline npm.js]: Note that this will only work properly given https://github.com/jspm/jspm-cli/issues/758.

File: lib/npmrc.js (hunk 2 above)
Comment (crisptrutski) [inline lib/npmrc.js]: No, it's just the legacy logic that was already here. Will add the extra paths when I add cascading

File: npm.js (hunk 3 above)
Comment (crisptrutski) [inline npm.js]: Ah, yes I was tripping over this. Thanks for pointing

**Comment by crisptrutski** [comment]
@guybedford think this is good-to-go, will pick up cascading tomorrow

**Comment by guybedford** [comment]
Sure.

**Pull request merged**